library/md_psa.h

```diff
@@ -0,0 +1,60 @@
+/**
+ * Translation between MD and PSA identifiers (algorithms, errors).
+ *
+ *  Note: this internal module will go away when everything becomes based on
+ *  PSA Crypto; it is a helper for the transition period.
+ *
+ *  Copyright The Mbed TLS Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+#ifndef MBEDTLS_MD_PSA_H
+#define MBEDTLS_MD_PSA_H
+
+#include "common.h"
+
+#include "mbedtls/md.h"
+#include "psa/crypto.h"
+
+/**
+ * \brief           This function returns the PSA algorithm identifier
+ *                  associated with the given digest type.
+ *
+ * \param md_type   The type of digest to search for.
+ *
+ * \return          The PSA algorithm identifier associated with \p md_type.
+ * \return          PSA_ALG_NONE if the algorithm is not supported.
```

Comment:
> some parts of the code are relying on "unsupported returns NONE"

From what I remember of prototyping what then became md-light, at least RSA relied on unsupported returning none (I remember finding out the hard way because it wasn't documented at the time). I haven't checked what the requirement is exactly though.

Anyway relaxing the condition wouldn't help the implementation now. I was wondering in case we decide to renumber algorithms so the translation could just be a bitmask, _if_ it doesn't have to account for unsupported algorithms precisely.

Reply:
> I remember finding out the hard way because it wasn't documented at the time

Yes, sorry for that. I think RSA is now using `md_info_from_type()` for that so it should no longer depend on that.

> Anyway relaxing the condition wouldn't help the implementation now. I was wondering in case we decide to renumber algorithms so the translation could just be a bitmask, if it doesn't have to account for unsupported algorithms precisely.

Absolutely, and this is why I was planning on looking at that as part of #6994 (which is that renumbering).